app/controllers/tasks_controller.rb

```diff
@@ -35,7 +35,7 @@ def show
     @tests = @task.tests
     @model_solutions = @task.model_solutions
 
-    @user_rating = @task.ratings&.find_by(user: current_user)&.rating
+    @user_rating = @task.ratings&.find_by(user: current_user) || Rating.new
```

Comment:
By the way: Do we need to handle incomplete tasks, where some aspects are missing? For example, when a task doesn't feature any test or model solution, users would still need to rate these. 

I think, there are two possibilities to conclude: Either, one argues that this is bad, since users would otherwise rate with 1 star only, which is still kept after adding a test / model solution (making it more difficult to examine). Or, one argues that it is fair to rate everything, since a missing element justifies a low rating for this category.

What do you think?

Reply:
Yeah, that's true. Let's encourage users to add all components here, too, and keep the current implementation 😉.